### PR TITLE
Fix handling of root directory as parent path

### LINKED ISSUE #14: False negative when parent path is the root directory
The code inside of this module assumes that `path.resolve(parentPath)` returns a path without a trailing slash, which is generally correct. However, the root directory is an exception that is not currently accounted for.

This results in a false negative:

```js
isPathInside('/foo', '/');  // false
```

### PR DESCRIPTION
**Pull request by sholladay**
Fixes #14

Before this PR:

```js
isPathInside('/foo', '/');  // false
```

After this PR:

```js
isPathInside('/foo', '/');  // true
```

*Note: commit https://github.com/sindresorhus/is-path-inside/pull/15/commits/b2bc9ea6c888aeb1fa913bb66aab9baf5a94e46f attempts to simplify the code by using `path.relative()`, which handles path resolution and case sensitivity internally. This change can be reverted safely and the original bug will still be fixed, if desired (e.g. for stylistic reasons).*

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,19 +2,11 @@
...
-	}
-
-	parentPath += parentPath.endsWith(path.sep) ? '' : path.sep;
-
-	return childPath.startsWith(parentPath);
+	const relation = path.relative(parentPath, childPath);
+	return Boolean(
+		relation &&
+		relation !== '..' &&
+		!relation.startsWith('..' + path.sep) &&
```
Comment (sindresorhus) [inline index.js]:
```suggestion
		!relation.startsWith(`..${path.sep}`) &&
```

**Comment by sindresorhus** [comment]
> Note: commit b2bc9ea attempts to simplify the code by using path.relative(), which handles path resolution and case sensitivity internally. This change can be reverted safely and the original bug will still be fixed, if desired (e.g. for stylistic reasons).

I like the change. I'm just paranoid that it will cause some unforeseen issues. For example, with some cases we haven't tested. Would you be able to add some more tests maybe?

File: index.js (hunk 1 above)
Comment (sholladay) [reply index.js]: Done. 👍 

Would be nice to codify that in XO. I find `${}` somewhat less readable than <code>&nbsp;+&nbsp;</code>, so I have a habit of only using template strings if it actually shortens the code, which it does in some cases, but not here.

**Comment by sholladay** [comment]
> Would you be able to add some more tests maybe?

Sure, what did you have in mind? Did you see the assertions I added in the first commit, https://github.com/sindresorhus/is-path-inside/pull/15/commits/bde790ba081d36751d5b8332ec1b2481fc650dfd? I tried to add all of the relevant permutations I could think of for this module.

**Comment by sindresorhus** [comment]
Maybe some longer paths and some variations of `../../`, `foo/../../bar`.

**Comment by sindresorhus** [comment]
@sholladay Friendly bump :)

**Comment by DerekZiemba** [comment]
Doesn't work on windows.  Try the following instead:

```js

module.exports = (childPath, parentPath) => {
	const p = process.platform === 'win32' ? path.win32 : path.posix;
	const relation = p.relative(parentPath, childPath);
	return Boolean(relation) && relation !== '..' && !relation.startsWith('..' + p.sep) && relation !== p.resolve(childPath);
};
```

I started looking into this repo for the following windows case:
```
t.true(isPathInside('\\\\?\\A:\\GitHub\\sindresorhus\\is-path-inside\\node_modules\\array-includes', 'a:\\github\\sindresorhus\\is-path-inside\\node_modules'));
t.true(isPathInside('\\\\?\\a:\\github\\sindresorhus\\is-path-inside\\node_modules\\array-includes', 'A:\\GitHub\\sindresorhus\\is-path-inside\\node_modules'));
```
I hoped the module would be able to handle win32 extended-length paths (it doesn't).

**Comment by sholladay** [comment]
Your win32 check is not necessary, the path module does that automatically when using `path.relative()`. I'll add some more Windows tests soon.

**Comment by sindresorhus** [comment]
Bump :)

**Comment by sholladay** [comment]
I've added some more assertions. IMO, the tests are in good shape, but I can add some more if necessary.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**